### ISSUE TITLE
AWS Lambda support for .NET Core 3.1

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#12610]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Lambda now supports .NET Core 3.1.

### New or Affected Resource(s)

* [`aws_lambda_function`](https://www.terraform.io/docs/providers/aws/r/lambda_function.html)
* [`aws_lambda_layer_version`](https://www.terraform.io/docs/providers/aws/r/lambda_layer_version.html)

### Potential Terraform Configuration

```hcl
resource "aws_lambda_function" "example" {
  runtime = "dotnetcore3.1"
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/03/aws-lambda-now-supports-net-core-3-1/).
[Blog post](https://aws.amazon.com/blogs/compute/announcing-aws-lambda-supports-for-net-core-3-1/).

Requires [AWS SDK v1.30.1](https://github.com/aws/aws-sdk-go/releases/tag/v1.30.1):
* https://github.com/terraform-providers/terraform-provider-aws/pull/12414

**Comment by Abbatss** [comment]
Hi @ewbankkit  ,
  I can see that you've added support for other runtimes previously.
Are you working on this issue?
If not, let me know, I'll create PR with fix. 
I want to deploy my lambda with new runtime as soon as possible.

**Comment by ewbankkit** [comment]
@Abbatss We'll need #12414 merged in order to pick up the new runtime constant value.

**Comment by bflad** [comment]
Support for this has been merged in and will release with version 2.57.0 of the Terraform AWS Provider, later this week. Thanks to @ewbankkit for the implementation. 👍
